Core.DomainModel/Organization/StsOrganizationConnection.cs

```diff
@@ -9,30 +12,82 @@ namespace Core.DomainModel.Organization
     /// </summary>
     public class StsOrganizationConnection : Entity, IOwnedByOrganization
     {
+        public StsOrganizationConnection()
+        {
+            StsOrganizationChangeLogs = new List<StsOrganizationChangeLog>();
+        }
+
         public int OrganizationId { get; set; }
         public virtual Organization Organization { get; set; }
         public bool Connected { get; set; }
         /// <summary>
         /// Determines the optional synchronization depth used during synchronization from STS Organisation
         /// </summary>
         public int? SynchronizationDepth { get; set; }
+        public virtual ICollection<StsOrganizationChangeLog> StsOrganizationChangeLogs { get; set; }
+
 
         public bool SubscribeToUpdates { get; set; }
-        //TODO https://os2web.atlassian.net/browse/KITOSUDV-3317 adds the change logs here
+        
         public DisconnectOrganizationFromOriginResult Disconnect()
         {
             var organizationUnits = Organization.OrgUnits.Where(x => x.Origin == OrganizationUnitOrigin.STS_Organisation).ToList();
             organizationUnits.ForEach(unit => unit.ConvertToNativeKitosUnit());
+            var removedLogs = RemoveAllLogs();
 
             Connected = false;
             SubscribeToUpdates = false;
             SynchronizationDepth = null;
-            return new DisconnectOrganizationFromOriginResult(organizationUnits);
+            return new DisconnectOrganizationFromOriginResult(organizationUnits, removedLogs);
         }
 
         public IExternalOrganizationalHierarchyUpdateStrategy GetUpdateStrategy()
         {
             return new StsOrganizationalHierarchyUpdateStrategy(Organization);
         }
+
+        public StsOrganizationConnectionAddNewLogsResult AddNewLogs(IEnumerable<StsOrganizationChangeLog> newLogs)
+        {
+            var newLogsList = newLogs.ToList();
+            foreach (var newLog in newLogsList)
+            {
+                StsOrganizationChangeLogs.Add(newLog);
+            }
+            var removedLogs = RemoveOldestLogs(newLogsList);
+
+            return new StsOrganizationConnectionAddNewLogsResult(removedLogs);
+        }
+
+        public Result<IEnumerable<StsOrganizationChangeLog>, OperationError> GetLastNumberOfChangeLogs(int number = StsOrganizationConnectionConstants.TotalNumberOfLogs)
+        {
+            if (number <= 0)
+            {
+                return new OperationError("Number of change logs to get cannot be lower than 0", OperationFailure.BadState);
```

Comment:
again... this error message disagrees with the parameter check... it MUST be above zero, not just "not lower than 0)